### PR TITLE
Die of fate tweaks and fixes

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
fix: Rolling a 6 with a die of fate now reduces your speed as intended.
fix: Rolling an 8 with a die of fate will cause the explosion to be
around the roller, not the die.
tweak: Die of fate effects now make loud visible messages so it's
obvious what has happened.
admin: Dice can now be "totally rigged" with admin edits to
unconditionally always roll a certain value, rather than just some of
the time. A new "cursed die of fate" has been added to demonstrate this
effect.
admin: Added the "stealth" die of fate, both one use and unlimited. You know,
for badminry.
/:cl:

- new proc `do_smoke` that does basic smoke effects, to avoid the same
pattern for making smoke effects.
- Dice rigging has been split into two vars, "rigged" for the severity
of the rigging (not rigged, basically rigged, totally rigged), and
rigged_value for the value it's trying to rig to.

### CODE REVIEW

**Comment by Time-Green** [comment]
Could you please make the resistence throw use physiology instead of species while you’re at it?

**Comment by AlexTheSysop42** [comment]
Make rolling a 20 drop all your items too, including the die if it's in your hand

**Comment by coiax** [comment]
Why would rolling the best result make you drop everything

**Comment by AlexTheSysop42** [comment]
Because otherwise it gets deleted

File: code/__DEFINES/misc.dm [hunk 1]
```diff
@@ -453,3 +453,7 @@ GLOBAL_LIST_INIT(pda_styles, list(MONO, VT, ORBITRON, SHARE))
 #define VARSET_FROM_LIST_IF(L, V, C...) if(L && L[#V] && (C)) V = L[#V]
 #define VARSET_TO_LIST(L, V) if(L) L[#V] = V
 #define VARSET_TO_LIST_IF(L, V, C...) if(L && (C)) L[#V] = V
+
+#define DICE_NOT_RIGGED "not-rigged"
```
Comment (Jordie0608) [inline code/__DEFINES/misc.dm]: Unless I've missed it these aren't used in strings and just for comparison, if so make them integers.

**Review by Jordie0608 — CHANGES_REQUESTED** [review]

**Pull request merged**